### PR TITLE
Fix #67 report doesn't print a diff when a value is a structure

### LINKED ISSUE #67: Report doesn't print a diff when a value is a structure

### PR DESCRIPTION
**Pull request by LukasRychtecky**

### CODE REVIEW

File: eftest/src/eftest/report/pretty.clj [hunk 1]
```diff
@@ -45,20 +44,16 @@
      (when (or file line)
        (str " (" (:source *fonts*) file ":" line (:reset *fonts*) ")")))))
 
-(defn- diff-all [expected actuals]
```
Comment (LukasRychtecky) [inline eftest/src/eftest/report/pretty.clj]: No more needed, a diff is in an event.

File: eftest/src/eftest/report/pretty.clj [hunk 2]
```diff
@@ -111,15 +106,14 @@
...
+(defmethod report :fail [{:keys [message expected diffs] :as m}]
   (test/with-test-out
     (test/inc-report-counter :fail)
     (print *divider*)
     (println (str (:fail *fonts*) "FAIL" (:reset *fonts*) " in") (testing-scope-str m))
     (when (seq test/*testing-contexts*) (println (test/testing-contexts-str)))
     (when message (println message))
-    (if (and (sequential? expected)
-             (= (first expected) '=))
+    (if (some? diffs)
```
Comment (LukasRychtecky) [inline eftest/src/eftest/report/pretty.clj]: There was a main issue.

**Comment by LukasRychtecky** [comment]
Hmm when I run tests on localhost it works 🤔 
```
Performing task 'eftest' with profile(s): 'base,system,user,provided,dev'

7/7   100% [==================================================]  ETA: 00:00

Ran 6 tests in 0.133 seconds
9 assertions, 0 failures, 0 errors.
Performing task 'eftest' with profile(s): 'base,system,user,provided,dev,1.8'

7/7   100% [==================================================]  ETA: 00:00

Ran 7 tests in 0.116 seconds
9 assertions, 0 failures, 0 errors.
Performing task 'eftest' with profile(s): 'base,system,user,provided,dev,1.9'

7/7   100% [==================================================]  ETA: 00:00

Ran 7 tests in 0.101 seconds
9 assertions, 0 failures, 0 errors.
Performing task 'eftest' with profile(s): 'base,system,user,provided,dev,1.10'

7/7   100% [==================================================]  ETA: 00:00

Ran 7 tests in 0.109 seconds
9 assertions, 0 failures, 0 errors.
```

**Comment by weavejester** [comment]
The tests fail on my machine as well as on the CI server. I'm looking through the source code of [clojure.test](https://github.com/clojure/clojure/blob/master/src/clj/clojure/test.clj), and nowhere is there a `:diffs` key. Where is this key coming from? Could you have a plugin that's adding it?

**Comment by LukasRychtecky** [comment]
> The tests fail on my machine as well as on the CI server. I'm looking through the source code of [clojure.test](https://github.com/clojure/clojure/blob/master/src/clj/clojure/test.clj), and nowhere is there a `:diffs` key. Where is this key coming from? Could you have a plugin that's adding it?

Heh. Good question, this is my `~/.lein/profiles.clj` I'm on Clojure `1.9.0`:
```clojure
{:user {:plugins [[lein-ancient "0.6.15"]
                  [venantius/ultra "0.5.2"]
                  [venantius/yagni "0.1.7"]
                  [lein-cloverage "1.0.11"]
                  [lein-nvd "0.6.0"]]}}}
```

Does it work for you on current stable version? I mean if you see the diff lines.

**Comment by weavejester** [comment]
Yes, I've just tested #67 and it works for me. My guess is that you have a plugin that is interferring. I'll update the original issue.

**Comment by Deraen** [comment]
`:diffs` is indeed from Ultra: https://github.com/venantius/ultra/blob/master/src/ultra/test.clj#L29

**Comment by LukasRychtecky** [comment]
> `:diffs` is indeed from Ultra: https://github.com/venantius/ultra/blob/master/src/ultra/test.clj#L29

Ah, my fault. Thx @Deraen. I think we can close it now, right? @weavejester 

Or does it make sense to make it Eftest & Ultra compatible?

**Comment by weavejester** [comment]
I think this is an issue with Ultra, rather than Eftest. Ultra redefines `assert-expr` for `=` and changes the type of `:actual` key from a single value into a list of values. Adding extra keys to the reporting map is fine; but changing existing keys means that Ultra is incompatible with anything that expects to be using `clojure.test`.

**Comment by weavejester** [comment]
Ah, actually after further investigation, I believe Ultra actually removes the `=` symbol from the start of the `:actual` key. So in clojure.test, it would be `(= 1 1)`, while in Ultra it would be `(1 1)`. Eftest looks for the beginning `=`.

**Pull request closed without merging**